### PR TITLE
[SDK] Modify DefaultPlacementUpdater not fail but skip old Configuration formats

### PR DESCRIPTION
**Pull request by mbdcos**
* Additions:
  - Do not give error if a task's config is not ServiceSpec
  - If a config is not ServiceSpec do not delete it
* Existing:
  - Delete duplicates configs
  - Delete unnecessary configs

### CODE REVIEW

File: sdk/scheduler/src/test/java/com/mesosphere/sdk/config/ConfigurationUpdaterTest.java [hunk 1]
```diff
@@ -133,6 +159,24 @@ public void testValidationDifferentConfigs() throws ConfigStoreException {
...
+                        DefaultScheduler.defaultConfigValidators());
+        when(mockConfigStore.getTargetConfig()).thenReturn(TARGET_ID);
+        when(mockConfigStore.fetch(TARGET_ID)).thenReturn(ORIGINAL_SERVICE_SPECIFICATION);
+        when(mockConfigStore.list()).thenReturn(Arrays.asList(TARGET_ID, NEW_ID, UNKNOWN_ID));
+        when(mockConfigStore.fetch(UNKNOWN_ID)).thenReturn(mockUnknownConfig);
+        ConfigurationUpdater.UpdateResult result = configurationUpdater.updateConfiguration(ORIGINAL_SERVICE_SPECIFICATION);
+        Assert.assertEquals(TARGET_ID, result.targetId);
+        Assert.assertTrue(result.errors.isEmpty());
+    }
+
```
Comment (gabrielhartmann) [inline sdk/scheduler/src/test/java/com/mesosphere/sdk/config/ConfigurationUpdaterTest.java]: extra newline

**Comment by mbdcos** [comment]
retest this please

**Comment by mbdcos** [comment]
retest this please

**Comment by mbdcos** [comment]
retest this please

**Comment by mbdcos** [comment]
Retest this please

**Pull request merged**